### DISCUSSION TITLE
Best practice for get session data from a React Server Component in Next.js 12

### DISCUSSION THREAD

**Discussion by KlotzJesse** [nextauthjs/next-auth#3308] [Help]
Hello,

is there currently any (best) practice or either a possiblity to access the session data from a react server component, as there is no connection to the client or even the client request?

Therefore requesting an API endpoint is useless as the server makes the request and therefore the request from API endpoint view doesn't contain any session data.

Thanks

**Comment by balazsorban44** [comment]
RSC in Next.js (and in general) is experimental at the moment, and I haven't had the chance to look into it yet.

When it gets more stable, I'll start exploring.

**Reply by balazsorban44** [reply]
Here is a roadmap https://github.com/vercel/next.js/discussions/31263
